# 10X Genomics scRNA-seq Analysis Pipeline

This pipeline performs primary and secondary analysis of 10X Genomics single-cell RNA-seq (scRNA-seq data). It includes the following steps:

#### 1. Primary analysis (Cell Ranger):
Preprocessing of raw FASTQ files generated by the 10X Genomics platform
Read quality assessment using FastQC and quality report generated with multiqc
Alignment of reads to a reference genome 
Gene quantification and UMI counting
Generation of count matrices for downstream analysis*

#### 2. Secondary analysis (Cell Ranger & Seurat) 
Data normalization
Dimensionality reduction & cell clustering
Differential gene expression analysis
Visualization

## Installation

Requirements:

Before running the pipeline, make sure that the following dependencies and tools are installed and included in your path:

1. Nextflow: See more in https://www.nextflow.io/docs/latest/overview.html
2. Cell ranger: See more in https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/installation
3. FastQC: See more in 
4. MultiQC: See more in https://multiqc.info/docs/getting_started/installation/ 
5. R version 4.3.1 or higher

### Installing dependencies

#### 1. Install Nextflow

1. Install Nextflow
```
curl -s https://get.nextflow.io | bash
```
2. Make Nextflow executable
```
chmod +x nextflow
```
3. Move Nextflow into an executable path:
```
sudo mv nextflow /usr/local/bin     
 ```
4. Confirm that Nextflow is installed correctly" 
```
nextflow info
```
#### 2. Install Cell Ranger

1. Download and unpack the cellranger `cellranger-x.y.z.tar.gz/.tar.xz` compressed folder from the 10xgenomics website: https://www.10xgenomics.com/support/software/cell-ranger/downloads

2. Add Cell Ranger to your PATH
```
export PATH=/file-location:$PATH
```

#### 3. Install FastQC 
Using conda:
```bash
conda install bioconda::fastqc
```

Using apt install:

```bash
`sudo apt-get install fastqc`
#or download from https://www.bioinformatics.babraham.ac.uk/projects/fastqc/
```

#### 4. Install MultiQC 
Using conda:

```bash
conda install multiqc
```

Using apt install:
```bash
sudo apt-get install multiqc
```

#### 5. Install R
Using conda: 
```bash
conda install r-base r-dev
```
Execute R and install the following packages: 
```bash
install.packages("Seurat")
install.packages("cowplot")
install.packages("ggplot2")
```

### Alternative: Create a conda environment to manage dependencies:

```bash
conda create -n scrna-seq-pipeline -c bioconda nextflow fastqc multiqc r-base r-dev r-Seurat r-cowplot 
conda activate scrna-seq-pipeline
```

# Preparing input files

### Input files

The FASTQ files must be located in the `data/reads` folder. Alternatively, the path to the reads-containing folder can be changed in the nextflow.config file, replacing the argument `reads = "$PWD/data/reads/_*R{1,2}_.fastq.gz"` with the path where the reads are located

### Naming your input FASTQ files
This pipeline requires the input files in .FASTQ format and following 10X naming convention:
`[Sample Name]_S1_L00[Lane Number]_[Read Type]_001.fastq.gz`

where ```Read Type``` is one of:
```bash
    I1: Sample index read (optional)
    I2: Sample index read (optional)
    R1: Read 1
    R2: Read 2
```

### Reference transcriptome

The reference transcriptome must be provided in the `data/ref` folder. The human and mice pre-built cellranger-compatible transcriptomes are available to download.

Human reference (GRCh38)-2024-A:
```bash
curl -O "https://cf.10xgenomics.com/supp/cell-exp/refdata-gex-GRCh38-2024-A.tar.gz"
```
Mouse reference (GRCm39)-2024-A:
```bash
curl -O "https://cf.10xgenomics.com/supp/cell-exp/refdata-gex-GRCm39-2024-A.tar.gz"
```

Custom reference transcriptomes (non-human, non-mouse) can be generated using a reference genome sequence and gene annotations. To do so, use the cellranger_mkref.nf module

# Running the pipeline

Run the following command to execute the pipeline:

```bash
nextflow run main.nf
```
